### ISSUE TITLE
Restore backup from 3.2 to 4.6 failed

### ISSUE THREAD

**Issue by patrickBoucly** [eXist-db/exist#2763]
What is the problem
A problem occured when i try to restore a backup zip from an 3.2 exist database to a 4.6 database

What did you expect
I expect the restoration success

Describe how to reproduce or add a test
The .zip backup file is done from the backup central (database size : 2Gb)
the steps i follow are:

```
chown -R exist: /opt/*
systemctl stop existdb.service
rm /var/lib/exist/data/*.log /var/lib/exist/data/*.dbx /var/lib/exist/data/*.lck
systemctl start existdb.service
su - exist -c "/var/lib/exist/bin/backup.sh -P PASSWORD_BACKUP_DB -r /opt/dir_backup/db/__contents__.xml"
```

it works perfectly on a new exist 3.2 database but failed on a 4.6 instance.
It return:

```
root@hostname:/var/lib/exist/tools/yajsw/work# su - exist -c "/var/lib/exist/bin/backup.sh -P PASSWORD_BACKUP_DB -r /opt/dir_backup/db/__contents__.xml"
-P PASSWORD_BACKUP_DB -r /opt/dir_backup/db/__contents__.xml
Using locale: fr_FR.UTF-8
mai 28, 2019 3:55:26 PM java.util.prefs.FileSystemPreferences$1 run
INFOS: Created user preferences directory.
Starting restore of backup...
Processing backup: /opt/dir_backup/db/system/security/__contents__.xml
Creating collection /db/system/security
An unrecoverable error occurred while restoring
collection '/db/system/security': HTTP server returned unexpected status: Unauthorized. Aborting restore!
Finished restore of backup.
An unrecoverable error occurred while restoring
collection '/db/system/security': HTTP server returned unexpected status: Unauthorized. Aborting restore!
------------------------------------
Problems occured found during restore:
WARN: An unrecoverable error occurred while restoring
collection '/db/system/security': HTTP server returned unexpected status: Unauthorized. Aborting restore!
ERROR: An unrecoverable error occurred while restoring
collection '/db/system/security': HTTP server returned unexpected status: Unauthorized. Aborting restore!
```

If you restored collections inside /db/apps, you may want
to rebuild the application repository. To do so, run the following query
as admin:

```xquery
import module namespace repair="http://exist-db.org/xquery/repo/repair"
at "resource:org/exist/xquery/modules/expathrepo/repair.xql";
repair:clean-all(),
repair:repair()
```

```
org.xmldb.api.base.XMLDBException: HTTP server returned unexpected status: Unauthorized
        at org.exist.xmldb.RemoteCollection.instance(RemoteCollection.java:99)
        at org.exist.xmldb.RemoteCollection.instance(RemoteCollection.java:75)
        at org.exist.xmldb.DatabaseImpl.readCollection(DatabaseImpl.java:267)
        at org.exist.xmldb.DatabaseImpl.getRemoteCollection(DatabaseImpl.java:229)
        at org.exist.xmldb.DatabaseImpl.getCollection(DatabaseImpl.java:169)
        at org.exist.xmldb.DatabaseImpl.getCollection(DatabaseImpl.java:158)
        at org.xmldb.api.DatabaseManager.getCollection(DatabaseManager.java:235)
        at org.exist.backup.Main.process(Main.java:261)
        at org.exist.backup.Main.main(Main.java:406)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.exist.start.Main.invokeMain(Main.java:140)
        at org.exist.start.Main.run(Main.java:453)
        at org.exist.start.Main.main(Main.java:64)
Caused by: org.apache.xmlrpc.client.XmlRpcHttpTransportException: HTTP server returned unexpected status: Unauthorized
        at org.apache.xmlrpc.client.XmlRpcSunHttpTransport.getInputStream(XmlRpcSunHttpTransport.java:94)
        at org.apache.xmlrpc.client.XmlRpcStreamTransport.sendRequest(XmlRpcStreamTransport.java:152)
        at org.apache.xmlrpc.client.XmlRpcHttpTransport.sendRequest(XmlRpcHttpTransport.java:143)
        at org.apache.xmlrpc.client.XmlRpcSunHttpTransport.sendRequest(XmlRpcSunHttpTransport.java:69)
        at org.apache.xmlrpc.client.XmlRpcClientWorker.execute(XmlRpcClientWorker.java:56)
        at org.apache.xmlrpc.client.XmlRpcClient.execute(XmlRpcClient.java:167)
        at org.apache.xmlrpc.client.XmlRpcClient.execute(XmlRpcClient.java:158)
        at org.apache.xmlrpc.client.XmlRpcClient.execute(XmlRpcClient.java:147)
        at org.exist.xmldb.RemoteCollection.instance(RemoteCollection.java:87)
        ... 15 more
caused by
org.apache.xmlrpc.client.XmlRpcHttpTransportException: HTTP server returned unexpected status: Unauthorized
        at org.apache.xmlrpc.client.XmlRpcSunHttpTransport.getInputStream(XmlRpcSunHttpTransport.java:94)
        at org.apache.xmlrpc.client.XmlRpcStreamTransport.sendRequest(XmlRpcStreamTransport.java:152)
        at org.apache.xmlrpc.client.XmlRpcHttpTransport.sendRequest(XmlRpcHttpTransport.java:143)
        at org.apache.xmlrpc.client.XmlRpcSunHttpTransport.sendRequest(XmlRpcSunHttpTransport.java:69)
        at org.apache.xmlrpc.client.XmlRpcClientWorker.execute(XmlRpcClientWorker.java:56)
        at org.apache.xmlrpc.client.XmlRpcClient.execute(XmlRpcClient.java:167)
        at org.apache.xmlrpc.client.XmlRpcClient.execute(XmlRpcClient.java:158)
        at org.apache.xmlrpc.client.XmlRpcClient.execute(XmlRpcClient.java:147)
        at org.exist.xmldb.RemoteCollection.instance(RemoteCollection.java:87)
        at org.exist.xmldb.RemoteCollection.instance(RemoteCollection.java:75)
        at org.exist.xmldb.DatabaseImpl.readCollection(DatabaseImpl.java:267)
        at org.exist.xmldb.DatabaseImpl.getRemoteCollection(DatabaseImpl.java:229)
        at org.exist.xmldb.DatabaseImpl.getCollection(DatabaseImpl.java:169)
        at org.exist.xmldb.DatabaseImpl.getCollection(DatabaseImpl.java:158)
        at org.xmldb.api.DatabaseManager.getCollection(DatabaseManager.java:235)
        at org.exist.backup.Main.process(Main.java:261)
        at org.exist.backup.Main.main(Main.java:406)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.exist.start.Main.invokeMain(Main.java:140)
        at org.exist.start.Main.run(Main.java:453)
        at org.exist.start.Main.main(Main.java:64)
```

Context information
- eXist-db 4.6.1
- Java version : openjdk version "1.8.0_171"
- Operating system : Debian 8.11
- 64 bit
- How is eXist-db installed? 
we use puppet to install exist. Source are deployed from exist.tar.bz2 and exist is install as a service. 

Thanks in advance for any help you are able to provide

**Comment by wolfgangmm** [comment]
Have you set an admin password on the new 4.6 database before the restore? If yes, you need to supply it with parameter `-p` in addition to the password used in the backup, which is specified by `-P`.

**Comment by patrickBoucly** [comment]
Thanks for your reply. Yes, i set a password on the 4.6 new database. I try to add it as you suggested but unfortunaly, i get the exact same error back

**Comment by StephanMa** [comment]
Is your admin password from the backup different then your current one? If so, try to start the restore process again and provide the former password.

**Comment by patrickBoucly** [comment]
they were different indeed. Making them identical don't solve the problem

**Comment by patrickBoucly** [comment]
Hello. I'm still stuck on this issue... Does anyone have another idea?  Thank you in advance for your support!

**Comment by craigberry** [comment]
Could you be more specific about what "making them identical" means?  The only thing I've ever seen work is ensuring that the admin password of the target eXist installation is the same as the admin password that was used to create the backup.  I usually have to do "bin/client.sh --no-gui" and then "passwd admin" in the new installation to set the password to whatever was used to create the backup.

If you specify both -p and -P on the backup command when doing the restore, it will work up until the point where the admin password from the security collection gets restored from the backup, which presumably overwrites the current value of that password, at which point subsequent operations will fail as unauthenticated.  At least that's what I infer from seeing restores that worked partway and then failed after some of the security database had been restored.

tl;dr -- before attempting the restore, set the password in the target instance to the same as what it was in the instance from which the backup was created.

**Comment by adamretter** [comment]
You could also try removing the admin account from the backup. Delete the file in the backup `/db/system/config/security/exist/users/admin.xml` and remove the entry for it from `/db/system/config/security/exist/users/__contents__.xml`

**Comment by StephanMa** [comment]
Not especially this issue related... but most of the time this does not work. The adminpassword is mostly changed even when i remove `/db/system` from `__contents__.xml`

**Comment by adamretter** [comment]
@StephanMa I recently updated the backup code and that will be released in eXist-db 5.0.0-RC8, hopefully that is no longer an issue for you.

**Comment by patrickBoucly** [comment]
@craigberry
I change the admin password on the exist 4.6 new installation, using the password of the DB i try to restore.
Then i execute:
chown -R exist: /opt/*
systemctl stop existdb.service
rm /var/lib/exist/data/*.log /var/lib/exist/data/*.dbx /var/lib/exist/data/*.lck
systemctl start existdb.service
su - exist -c "/var/lib/exist/bin/backup.sh -u admin -p PWD -P PWD -r /opt/dir_backup/db/__contents__.xml"
OR
su - exist -c "/var/lib/exist/bin/backup.sh -P PWD-r /opt/dir_backup/db/__contents__.xml"

Boths give me the same error as i my 1st message.

I finally find a solution, adding "ouri" and executing ./backup.sh with the service stopped:

With same password on the 4.6 DB and the backuped DB,
chown -R exist: /opt/*
systemctl stop existdb.service
rm /var/lib/exist/data/*.log /var/lib/exist/data/*.dbx /var/lib/exist/data/*.lck
su - exist -c "/var/lib/exist/bin/backup.sh -P PWD -ouri=xmldb:exist:// -r /opt/dir_backup/db/__contents__.xml"
systemctl start existdb.service

I don't know why the backup steps i follow in a 1st place works fined on 3.2 and didn't works on 4.6...
@adamretter : I didn't try yet this solution

Thanks everyone for your help!
